tests/test_pending_points.py

```diff
@@ -0,0 +1,145 @@
+"""Tests pending points mechanism."""
+
+import warnings
+
+import pandas as pd
+import pytest
+from pytest import param
+
+from baybe.acquisition.base import AcquisitionFunction
+from baybe.exceptions import IncompatibleAcquisitionFunctionError, UnusedObjectWarning
+from baybe.recommenders import (
+    BotorchRecommender,
+    FPSRecommender,
+    GaussianMixtureClusteringRecommender,
+    KMeansClusteringRecommender,
+    NaiveHybridSpaceRecommender,
+    PAMClusteringRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.utils.basic import get_subclasses
+from baybe.utils.dataframe import add_fake_results, add_parameter_noise
+from baybe.utils.random import temporary_seed
+
+_discrete_params = ["Categorical_1", "Switch_1", "Num_disc_1"]
+_continuous_params = ["Conti_finite1", "Conti_finite2", "Conti_finite3"]
+_hybrid_params = ["Categorical_1", "Num_disc_1", "Conti_finite1", "Conti_finite2"]
+
+# Repeated recommendations explicitly need to be allowed or the potential overlap will
+# be avoided trivially
+_flags = dict(
+    allow_repeated_recommendations=True, allow_recommending_already_measured=True
+)
+
+
+@pytest.mark.parametrize(
+    "parameter_names, recommender",
+    [
+        param(_discrete_params, FPSRecommender(**_flags), id="fps_discrete"),
+        param(_discrete_params, PAMClusteringRecommender(**_flags), id="pam_discrete"),
+        param(
+            _discrete_params,
+            KMeansClusteringRecommender(**_flags),
+            id="kmeans_discrete",
+        ),
+        param(
+            _discrete_params,
+            GaussianMixtureClusteringRecommender(**_flags),
+            id="gm_discrete",
+        ),
+        param(
+            _discrete_params,
+            TwoPhaseMetaRecommender(recommender=BotorchRecommender(**_flags)),
+            id="botorch_discrete",
+        ),
+        param(
+            _continuous_params,
+            TwoPhaseMetaRecommender(recommender=BotorchRecommender(**_flags)),
+            id="botorch_continuous",
+        ),
+        param(
+            _hybrid_params,
+            TwoPhaseMetaRecommender(recommender=BotorchRecommender(**_flags)),
+            id="botorch_hybrid",
+        ),
+        param(
+            _discrete_params,
+            NaiveHybridSpaceRecommender(
+                disc_recommender=FPSRecommender(**_flags), **_flags
+            ),
+            id="naive1_discrete",
+        ),
+        param(
+            _discrete_params,
+            NaiveHybridSpaceRecommender(
+                disc_recommender=KMeansClusteringRecommender(**_flags), **_flags
+            ),
+            id="naive2_discrete",
+        ),
+    ],
+)
+@pytest.mark.parametrize("n_grid_points", [8], ids=["grid8"])
+def test_pending_points(campaign, batch_size):
```

Comment:
This test does not check what happens if we then finally add the measurements for the pending points.

Reply:
You seem to have a special expectation about that case.
What is that expectation and why/what should it be the outcome?

Reply:
I don't expect anything special to happen, but it is something that just needs to be tested imo. *The* important part of having pending points is that these will also be added at a later point in time, and this is something that simply is not tested/does not happen anywhere. In my opinion, this needs to be added as we should at least have tests for this. But if you and @AdrianSosic (whom I summon here for his opinion) both think this is not necessary, feel free to resolve.

Reply:
Sorry, still dont understand. 
You want a test. But have no expectation as to what should be the oputcome of the test. Please tell me what is the outcome of the thing you want to be tested.

I am asking because adding a measurement previously marked as pending has no special role or effect, in requesting this being added you seem to imply otherwise. The important thing is not what happens when previously pending points are added, in fact this PR does nothing that even affects this situation --> hence I dont see what you want tested here

Reply:
@AVHopp requiring your response or resolution